ui-common/src/components/modals/TrafficStatsModal/TrafficStatsModal.tsx

```diff
@@ -0,0 +1,86 @@
+import React, {useEffect, useState} from "react";
+import {Backdrop, Box, Fade, Modal} from "@mui/material";
+import styles from "./TrafficStatsModal.module.sass";
+import closeIcon from "assets/close.svg";
+import {TrafficPieChart} from "./TrafficPieChart/TrafficPieChart";
+import spinnerImg from "assets/spinner.svg";
+
+const modalStyle = {
+  position: 'absolute',
+  top: '6%',
+  left: '50%',
+  transform: 'translate(-50%, 0%)',
+  width: '50vw',
+  height: '82vh',
+  bgcolor: 'background.paper',
+  borderRadius: '5px',
+  boxShadow: 24,
+  p: 4,
+  color: '#000',
+};
+
+enum StatsMode {
+  REQUESTS = "entriesCount",
+  VOLUME = "volumeSizeBytes"
+}
+
+interface TrafficStatsModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  getTrafficStatsDataApi: () => Promise<any>
+}
+
+export const TrafficStatsModal: React.FC<TrafficStatsModalProps> = ({ isOpen, onClose, getTrafficStatsDataApi }) => {
+
+  const modes = Object.keys(StatsMode).filter(x => !(parseInt(x) >= 0));
```

Comment:
Why do you need to filter StatsMode like that?

Reply:
Object.keys of an enum return the string keys but also return numerical keys.
for example if we have the enum {a: a, b: b}
Object.keys(enum) will return [a,b,0,1] and we want to filter those numbers

https://www.crojach.com/blog/2019/2/6/getting-enum-keys-in-typescript

Reply:
ok got it. thanks